bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/widgets/Button.java

```diff
@@ -528,62 +530,62 @@ private boolean isCheckButton() {
 	}
 
 	private void drawPushButton(GC gc, int x, int y, int w,
-			int h) {
+			int h, ColorProvider colorProvider) {
```

Comment:
Since the color provider is directly accessible everywhere in the controls, it is not necessary to pass it as a method argument.